### ISSUE TITLE
As of wgpu-rs 0.9.0 "format R32Float is not blendable"

### ISSUE THREAD

**Issue by Craig-Macomber** [gfx-rs/wgpu#1682]
**Description**

Despite having hardware capable of filtering and blending R32Float, and opting into wgpu::Features::TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES, I get a validation error when trying to blend R32Float.'

```
[0.098746 ERROR]()(no module): wgpu error: Validation Error

Caused by:
    In Device::create_render_pipeline
      note: label = `AutoSized`
    color state [0] is invalid
    format R32Float is not blendable


thread 'main' panicked at 'Handling wgpu errors as fatal by default', /home/craig/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-0.9.0/src/backend/direct.rs:1978:5
```

**Repro steps**

run create_render_pipeline that blends an R32Float texture, for example:

```rust
 let pipeline = device.create_render_pipeline(&wgpu::RenderPipelineDescriptor {
        label: Some(&accumulate.name),
        layout: Some(&pipeline_layout),
        vertex: vertex_shader,
        fragment: Some(wgpu::FragmentState {
            module: &data.shader,
            entry_point: if smaller.is_some() {
                "fs_main_textured"
            } else {
                "fs_main"
            },
            targets: &[wgpu::ColorTargetState {
                format: TextureFormat::R32Float,
                blend: Some(blend_state_add),
                write_mask: wgpu::ColorWrite::ALL,
            }],
        }),
        primitive: wgpu::PrimitiveState::default(),
        depth_stencil: None,
        multisample: wgpu::MultisampleState::default(),
    });
```
(This is an except from https://github.com/Craig-Macomber/rusty-flame/tree/brokenBlending which contains a repro for this issue, at least on my system)

I haven't made a more minimal repro since this requires shaders and such, but if making a more minimal repro would aid in debugging/fixing this let me know and I'll make one.

**Expected vs observed behavior**

I expect to be allowed to blend R32Float when using TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES on hardware that is capable of it.

I observe the error listed above which says it it not supported.

**Extra materials**
My setup claims:
```
[src/main.rs:103] adapter.get_texture_format_features(wgpu::TextureFormat::R32Float) = TextureFormatFeatures {
    allowed_usages: COPY_SRC | COPY_DST | SAMPLED | STORAGE | RENDER_ATTACHMENT,
    flags: STORAGE_READ_WRITE | STORAGE_ATOMICS,
    filterable: true,
}
```
I check that immediately before constructing the device from that adapter. Later when trying to create the render pipeline it failed.

https://github.com/gfx-rs/wgpu/pull/1647 implies blend-ability should match filter-ability, and this reports filterable, so it should be blendable as well.

**Platform**
Linux (Ubuntu 20.04)
Fails on wgpu 0.9.0. Confirmed NOT to be an issue in wgpu 0.8.1 (which appears to not validate this case at all, allowing it even when TEXTURE_ADAPTER_SPECIFIC_FORMAT_FEATURES is not provided)
GPU: Advanced Micro Devices, Inc. [AMD/ATI] Vega 10 XL/XT [Radeon RX Vega 56/64] (rev c3)

**Comment by kvark** [comment]
This was fixed in #1528, right after the release, and didn't make it to any patches yet.
Is this a blocker for you? I.e. can you use R16F for a workaround?

**Comment by Craig-Macomber** [comment]
> This was fixed in #1528, right after the release, and didn't make it to any patches yet.
> Is this a blocker for you? I.e. can you use R16F for a workaround?

Excellent.

Knowing that its fixed, I can just wait for the next release or use F16: F16 would get me ~5 bit color instead of 8 (I care mainly about the exponent bits): I can do that if I need to.

**Comment by kvark** [comment]
This is now published in wgpu-core-0.9.2

**Comment by Craig-Macomber** [comment]
Confirmed fixed with wgpu-core0.9.2. Thanks!
